### PR TITLE
rnndb updates for GLES3 on GC7000L

### PR DESCRIPTION
**Pull request by flto**
Making a pull request since there are many changes and you might have some suggestions about naming/etc.

Might have missed some changes (I have a bunch of uncommited GLES3 hacks with hardcoded values..), will update this PR if needed. Might also have new bits for GC3000  (I will try GLES3 stuff on GC3000 eventually).

### CODE REVIEW

File: rnndb/state.xml [hunk 1]
```diff
@@ -86,11 +86,14 @@ xsi:schemaLocation="http://nouveau.freedesktop.org/ rules-ng.xsd">
         <value value="3" name="UNSIGNED_SHORT"/>
         <value value="4" name="INT"/>
         <value value="5" name="UNSIGNED_INT"/>
+        <value value="6" name="INT_10_10_10_2"/>
```
Comment (austriancoder) [inline rnndb/state.xml]: I think these values have the following meaning (at least gc3000).
6: INT_2_10_10_10_REV
7: UNSIGNED_INT_2_10_10_10_REV

With that you should leave 12 and 13 untouched.

Comment (flto) [reply rnndb/state.xml]: What does the REV mean in this case?

Comment (austriancoder) [reply rnndb/state.xml]: reversed - regarding how the packed data is stored.

File: rnndb/state_3d.xml [hunk 2]
```diff
@@ -511,10 +525,28 @@ xsi:schemaLocation="http://nouveau.freedesktop.org/ rules-ng.xsd">
...
+                      should contain this information??
+                -->
+                <value value="1" name="A2B10G10R10UI"/>
+                <value value="2" name="UIF32"/>
+                <value value="3" name="U8"/>
+                <value value="4" name="U16"/>
+                <value value="5" name="S8"/>
+                <value value="6" name="S16"/>
+            </bitfield>
+            <!-- repeated every 4 bits (8 times for each RT) -->
```
Comment (laanwj) [inline rnndb/state_3d.xml]: If this repeats 8 times, let's make a `<bitfield>` entry for every instance and put the values in a separate enum definition.

File: rnndb/state_3d.xml [hunk 3]
```diff
@@ -810,12 +842,12 @@ xsi:schemaLocation="http://nouveau.freedesktop.org/ rules-ng.xsd">
             <bitfield pos="23" name="UNK20_MASK"/>
             <bitfield high="26" low="24" name="UNK24"/> <!-- Sampling related (set on GC3000 but not GC2000) -->
             <bitfield pos="27" name="UNK24_MASK"/>
-            <bitfield pos="30" name="UNK31_MASK"/>
-            <bitfield pos="31" name="UNK31"/>
+            <bitfield pos="30" name="SRGB_MASK"/>
+            <bitfield pos="31" name="SRGB"/> <!-- linear to sRGB convert output -->
```
Comment (laanwj) [inline rnndb/state_3d.xml]: Ah yes I suspected this but wasn't sure, thanks for verifying

File: rnndb/state.xml (hunk 1 above)
Comment (flto) [reply rnndb/state.xml]: I see, this matches the names of the types accepted by `glVertexAttribPointer` in GLES3. Changed it.

File: rnndb/state_3d.xml (hunk 2 above)
Comment (flto) [reply rnndb/state_3d.xml]: Changed this and also changed the names to from `S8/16` to `I8/16`

**Comment by flto** [comment]
Added a new commit about ASTC

**Comment by austriancoder** [comment]
lgtm

**Comment by laanwj** [comment]
> lgtm

Agree! Thanks for this!

**Pull request merged**